src/luma.c

```diff
@@ -95,21 +88,21 @@ void VS_CC lumaCreate(const VSMap *in, VSMap *out, void *userData, VSCore *core,
    d.node = vsapi->propGetNode(in, "clip", 0, 0);
    d.vi = *vsapi->getVideoInfo(d.node);
 
-
-   // In this first version we only want to handle 8 and 16bit integer formats. Note that
-   // vi->format can be 0 if the input clip can change format midstream.
-   if (!d.vi.format || d.vi.format->sampleType != stInteger || (d.vi.format->bitsPerSample != 8 && d.vi.format->bitsPerSample != 16)) {
-      vsapi->setError(out, "Luma: only constant format 8 and 16bit integer input supported");
+   if (!d.vi.format || d.vi.format->sampleType != stInteger || d.vi.format->bitsPerSample > 16) {
+      vsapi->setError(out, "Luma: only constant format 8 to 16 bit integer input supported");
       vsapi->freeNode(d.node);
       return;
    }
 
    // We don't need any chroma.
-   if(d.vi.format->bitsPerSample == 8)
-      d.vi.format = vsapi->getFormatPreset(pfGray8, core);
-   else
-      d.vi.format = vsapi->getFormatPreset(pfGray16, core);
-   
+   switch (d.vi.format->bitsPerSample) {
+   case 8: d.vi.format = vsapi->getFormatPreset(pfGray8, core); break;
+   case 16: d.vi.format = vsapi->getFormatPreset(pfGray16, core); break;
+   default: d.vi.format = vsapi->registerFormat(cmGray, stInteger, d.vi.format->bitsPerSample, 0, 0, core); break;
+   }
+
+   //Pow should be fine here since we only use it once.
+   d.maxVal = pow(2, d.vi.format->bitsPerSample) - 1;
```

Comment:
Is this is the only reason math.h is included? Powers of two are really easy with integer math:
```C
d.maxVal = (1 << d.vi.format->bitsPerSample) - 1;
```